### PR TITLE
Publish documentation.

### PR DESCRIPTION
**Pull request by jeschkies**
Summary:
This migrates the CI to the declarative Jenkins pipeline and publishes
the documentation generated by sbt-site. The documentation live in
their own `docs` sbt project. The API documentation is generated but
not yet included.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -4,23 +4,37 @@
...
-        sh 'sbt +publish'
+
+    stage("Publish") {
+      agent {
+        label "jdk8-scala"
       }
-    }}
+
+      environment {
+        AWS_ACCESS_KEY_ID = credentials('3f0dbb48-de33-431f-b91c-2366d2f0e1cf')
```
Comment (kaiwalyajoshi) [inline Jenkinsfile:31]: Can we remove these hardcoded ones?

**Review by kaiwalyajoshi — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (jeschkies) [reply Jenkinsfile:31]: This is not a hard coded key but a Jenkins credentials ID.

**Review by kaiwalyajoshi — APPROVED** [review]

File: build.sbt [hunk 2]
```diff
@@ -34,7 +35,35 @@ val commonSettings = Seq(
...
+    s3credentials := DefaultAWSCredentialsProviderChain.getInstance(),
+    s3region :=  com.amazonaws.services.s3.model.Region.US_Standard,
+    publish / skip := true,
+
+    name := "USI - Unified Scheduler Interface",
+    gitRemoteRepo := "https://github.com/mesosphere/usi",
+    ghpagesNoJekyll := true,
+
+    ParadoxMaterialThemePlugin.paradoxMaterialThemeSettings(Paradox),
+    // TODO: These settings do not work yet
```
Comment (jeschkies) [inline build.sbt:56]: @timcharper, would you have an idea why the following settings have to effect?

**Review by jeschkies — COMMENTED** [review]
I decided to create a `docs` project. It works quite nice. Just call `sbt docs/previewSite` and you can view any changes. However, publishing the docs does not work for some reasons from Jenkins.

**Comment by timcharper** [comment]
Exit status 1 when running "sbt +test:compile"

<details>
<summary>Details</summary>
<pre>
-Xmx2048M
-Xss6M
-XX:MaxPermSize=512M
-jar
/home/travis/.sbt/launchers/1.3.3/sbt-launch.jar
+test:compile

saving stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=512M; support was removed in 8.0
[info] Loading settings for project usi-build from plugins.sbt ...
[info] Loading project definition from /home/travis/build/mesosphere/usi/project
[warn] There may be incompatibilities among your library dependencies; run &#39;evicted&#39; to see detailed eviction warnings.
[info] Compiling 1 Scala source to /home/travis/build/mesosphere/usi/project/target/scala-2.12/sbt-1.0/classes ...
[info] Done compiling.
[info] Loading settings for project root from build.sbt ...
[info] Resolving key references (11443 settings) ...
[error] java.net.URISyntaxException: Illegal character in scheme name at index 3: git@github.com:mesosphere/usi
[error] Use &#39;last&#39; for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "sbt scalafmtCheck test:scalafmtCheck"

<details>
<summary>Details</summary>
<pre>
java
-Xms2048M
-Xmx2048M
-Xss6M
-XX:MaxPermSize=512M
-jar
/home/travis/.sbt/launchers/1.3.3/sbt-launch.jar
scalafmtCheck
test:scalafmtCheck

saving stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=512M; support was removed in 8.0
[info] Loading settings for project usi-build from plugins.sbt ...
[info] Loading project definition from /home/travis/build/mesosphere/usi/project
[info] Loading settings for project root from build.sbt ...
[info] Resolving key references (11443 settings) ...
[error] java.net.URISyntaxException: Illegal character in scheme name at index 3: git@github.com:mesosphere/usi
[error] Use &#39;last&#39; for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "Unused imports / scalafix"

<details>
<summary>Details</summary>
<pre>
java
-Xms2048M
-Xmx2048M
-Xss6M
-XX:MaxPermSize=512M
-jar
/home/travis/.sbt/launchers/1.3.3/sbt-launch.jar
&quot;scalafix --check&quot;
&quot;test:scalafix --check&quot;

saving stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=512M; support was removed in 8.0
[info] Loading settings for project usi-build from plugins.sbt ...
[info] Loading project definition from /home/travis/build/mesosphere/usi/project
[info] Loading settings for project root from build.sbt ...
[info] Resolving key references (11443 settings) ...
[error] java.net.URISyntaxException: Illegal character in scheme name at index 3: git@github.com:mesosphere/usi
[error] Use &#39;last&#39; for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "Running tests for all Scala versions"

<details>
<summary>Details</summary>
<pre>
# Executing command line:
java
-Xms2048M
-Xmx2048M
-Xss6M
-XX:MaxPermSize=512M
-jar
/home/travis/.sbt/launchers/1.3.3/sbt-launch.jar
+test

saving stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=512M; support was removed in 8.0
[info] Loading settings for project usi-build from plugins.sbt ...
[info] Loading project definition from /home/travis/build/mesosphere/usi/project
[info] Loading settings for project root from build.sbt ...
[info] Resolving key references (11443 settings) ...
[error] java.net.URISyntaxException: Illegal character in scheme name at index 3: git@github.com:mesosphere/usi
[error] Use &#39;last&#39; for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "sbt +test:compile"

<details>
<summary>Details</summary>
<pre>
-Xmx2048M
-Xss6M
-XX:MaxPermSize=512M
-jar
/home/travis/.sbt/launchers/1.3.3/sbt-launch.jar
+test:compile

saving stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=512M; support was removed in 8.0
[info] Loading settings for project usi-build from plugins.sbt ...
[info] Loading project definition from /home/travis/build/mesosphere/usi/project
[warn] There may be incompatibilities among your library dependencies; run &#39;evicted&#39; to see detailed eviction warnings.
[info] Compiling 1 Scala source to /home/travis/build/mesosphere/usi/project/target/scala-2.12/sbt-1.0/classes ...
[info] Done compiling.
[info] Loading settings for project root from build.sbt ...
[info] Resolving key references (11443 settings) ...
[error] java.net.URISyntaxException: Illegal character in scheme name at index 3: git@github.com:mesosphere/usi
[error] Use &#39;last&#39; for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "sbt scalafmtCheck test:scalafmtCheck"

<details>
<summary>Details</summary>
<pre>
java
-Xms2048M
-Xmx2048M
-Xss6M
-XX:MaxPermSize=512M
-jar
/home/travis/.sbt/launchers/1.3.3/sbt-launch.jar
scalafmtCheck
test:scalafmtCheck

saving stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=512M; support was removed in 8.0
[info] Loading settings for project usi-build from plugins.sbt ...
[info] Loading project definition from /home/travis/build/mesosphere/usi/project
[info] Loading settings for project root from build.sbt ...
[info] Resolving key references (11443 settings) ...
[error] java.net.URISyntaxException: Illegal character in scheme name at index 3: git@github.com:mesosphere/usi
[error] Use &#39;last&#39; for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "Unused imports / scalafix"

<details>
<summary>Details</summary>
<pre>
java
-Xms2048M
-Xmx2048M
-Xss6M
-XX:MaxPermSize=512M
-jar
/home/travis/.sbt/launchers/1.3.3/sbt-launch.jar
&quot;scalafix --check&quot;
&quot;test:scalafix --check&quot;

saving stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=512M; support was removed in 8.0
[info] Loading settings for project usi-build from plugins.sbt ...
[info] Loading project definition from /home/travis/build/mesosphere/usi/project
[info] Loading settings for project root from build.sbt ...
[info] Resolving key references (11443 settings) ...
[error] java.net.URISyntaxException: Illegal character in scheme name at index 3: git@github.com:mesosphere/usi
[error] Use &#39;last&#39; for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "Running tests for all Scala versions"

<details>
<summary>Details</summary>
<pre>
# Executing command line:
java
-Xms2048M
-Xmx2048M
-Xss6M
-XX:MaxPermSize=512M
-jar
/home/travis/.sbt/launchers/1.3.3/sbt-launch.jar
+test

saving stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=512M; support was removed in 8.0
[info] Loading settings for project usi-build from plugins.sbt ...
[info] Loading project definition from /home/travis/build/mesosphere/usi/project
[info] Loading settings for project root from build.sbt ...
[info] Resolving key references (11443 settings) ...
[error] java.net.URISyntaxException: Illegal character in scheme name at index 3: git@github.com:mesosphere/usi
[error] Use &#39;last&#39; for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "sbt +test:compile"

<details>
<summary>Details</summary>
<pre>
-Xmx2048M
-Xss6M
-XX:MaxPermSize=512M
-jar
/home/travis/.sbt/launchers/1.3.3/sbt-launch.jar
+test:compile

saving stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=512M; support was removed in 8.0
[info] Loading settings for project usi-build from plugins.sbt ...
[info] Loading project definition from /home/travis/build/mesosphere/usi/project
[warn] There may be incompatibilities among your library dependencies; run &#39;evicted&#39; to see detailed eviction warnings.
[info] Compiling 1 Scala source to /home/travis/build/mesosphere/usi/project/target/scala-2.12/sbt-1.0/classes ...
[info] Done compiling.
[info] Loading settings for project root from build.sbt ...
[info] Resolving key references (11443 settings) ...
[error] java.net.URISyntaxException: Illegal character in scheme name at index 3: git@github.com:mesosphere/usi
[error] Use &#39;last&#39; for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "sbt scalafmtCheck test:scalafmtCheck"

<details>
<summary>Details</summary>
<pre>
java
-Xms2048M
-Xmx2048M
-Xss6M
-XX:MaxPermSize=512M
-jar
/home/travis/.sbt/launchers/1.3.3/sbt-launch.jar
scalafmtCheck
test:scalafmtCheck

saving stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=512M; support was removed in 8.0
[info] Loading settings for project usi-build from plugins.sbt ...
[info] Loading project definition from /home/travis/build/mesosphere/usi/project
[info] Loading settings for project root from build.sbt ...
[info] Resolving key references (11443 settings) ...
[error] java.net.URISyntaxException: Illegal character in scheme name at index 3: git@github.com:mesosphere/usi
[error] Use &#39;last&#39; for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "Unused imports / scalafix"

<details>
<summary>Details</summary>
<pre>
java
-Xms2048M
-Xmx2048M
-Xss6M
-XX:MaxPermSize=512M
-jar
/home/travis/.sbt/launchers/1.3.3/sbt-launch.jar
&quot;scalafix --check&quot;
&quot;test:scalafix --check&quot;

saving stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=512M; support was removed in 8.0
[info] Loading settings for project usi-build from plugins.sbt ...
[info] Loading project definition from /home/travis/build/mesosphere/usi/project
[info] Loading settings for project root from build.sbt ...
[info] Resolving key references (11443 settings) ...
[error] java.net.URISyntaxException: Illegal character in scheme name at index 3: git@github.com:mesosphere/usi
[error] Use &#39;last&#39; for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "Running tests for all Scala versions"

<details>
<summary>Details</summary>
<pre>
# Executing command line:
java
-Xms2048M
-Xmx2048M
-Xss6M
-XX:MaxPermSize=512M
-jar
/home/travis/.sbt/launchers/1.3.3/sbt-launch.jar
+test

saving stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
OpenJDK 64-Bit Server VM warning: ignoring option MaxPermSize=512M; support was removed in 8.0
[info] Loading settings for project usi-build from plugins.sbt ...
[info] Loading project definition from /home/travis/build/mesosphere/usi/project
[info] Loading settings for project root from build.sbt ...
[info] Resolving key references (11443 settings) ...
[error] java.net.URISyntaxException: Illegal character in scheme name at index 3: git@github.com:mesosphere/usi
[error] Use &#39;last&#39; for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? 
restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "Running tests for all Scala versions"

<details>
<summary>Details</summary>
<pre>
[info] Total number of tests run: 0
[info] Suites: completed 0, aborted 1
[info] Tests: succeeded 0, failed 0, canceled 0, ignored 0, pending 0
[info] *** 1 SUITE ABORTED ***
[error] Error during tests:
[error] 	com.mesosphere.mesos.MasterDetectorTest
[info] Run completed in 47 seconds, 345 milliseconds.
[info] Total number of tests run: 8
[info] Suites: completed 2, aborted 0
[info] Tests: succeeded 8, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Run completed in 53 seconds, 731 milliseconds.
[info] Total number of tests run: 1
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 1, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[error] (mesos-master-detector / Test / test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 77 s (01:17), completed Apr 28, 2020 8:28:49 AM

restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by jeschkies** [comment]
We have some SVGs in the docs that should not be removed.

**Comment by timcharper** [comment]
Exit status 1 when running "Running tests for all Scala versions"

<details>
<summary>Details</summary>
<pre>
[info] Total number of tests run: 73
[info] Suites: completed 11, aborted 0
[info] Tests: succeeded 72, failed 1, canceled 0, ignored 0, pending 0
[info] *** 1 TEST FAILED ***
[error] Failed tests:
[error] 	com.mesosphere.usi.core.FlowHelpersTest
[info] Run completed in 1 minute, 25 seconds.
[info] Total number of tests run: 8
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 8, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Run completed in 1 minute, 18 seconds.
[info] Total number of tests run: 8
[info] Suites: completed 2, aborted 0
[info] Tests: succeeded 8, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[error] (core / Test / test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 108 s (01:48), completed Apr 28, 2020 4:02:59 PM

restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Comment by timcharper** [comment]
Exit status 1 when running "Building documentation"

<details>
<summary>Details</summary>
<pre>
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:281)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
[error] 	at sbt.Execute.work(Execute.scala:290)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:281)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error] 	at java.lang.Thread.run(Thread.java:748)
[error] (docs / Paradox / paradox) Duplicate mappings:
[error] 	/home/travis/build/mesosphere/usi/docs/target/paradox/paradox/site/paradox/design/index.html
[error] from
[error] 	/home/travis/build/mesosphere/usi/docs/src/paradox/design/index.html
[error] 		/home/travis/build/mesosphere/usi/docs/target/paradox/paradox/html/design/index.html
[error] Total time: 3 s, completed Apr 28, 2020 4:12:07 PM

restoring stty: 500:5:bf:8a3b:3:1c:7f:15:4:0:1:0:11:13:1a:0:12:f:17:16:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0
</pre>
</details>

**Pull request merged**